homeassistant/components/awair/__init__.py

```diff
@@ -74,9 +104,33 @@ async def _async_update_data(self) -> Any | None:
             except Exception as err:
                 raise UpdateFailed(err) from err
 
-    async def _fetch_air_data(self, device):
-        """Fetch latest air quality data."""
-        LOGGER.debug("Fetching data for %s", device.uuid)
-        air_data = await device.air_data_latest()
-        LOGGER.debug(air_data)
-        return AwairResult(device=device, air_data=air_data)
+
+class AwairLocalDataUpdateCoordinator(AwairDataUpdateCoordinator):
+    """Define a wrapper class to update Awair data from the local API."""
+
+    _device: AwairLocalDevice | None = None
+
+    def __init__(self, hass, config_entry, session) -> None:
+        """Set up the AwairLocalDataUpdateCoordinator class."""
+        self._awair = AwairLocal(
+            session=session, device_addrs=[config_entry.data[CONF_HOST]]
+        )
+
+        super().__init__(hass, config_entry, UPDATE_INTERVAL_LOCAL)
+
+    async def _async_update_data(self) -> dict[str, AwairResult] | None:
+        """Update data via Awair client library."""
+        async with timeout(API_TIMEOUT):
+            try:
+                if self._device is None:
+                    LOGGER.debug("Fetching devices")
+                    devices = await self._awair.devices()
+                    self._device = devices[0]
+                result = await self._fetch_air_data(self._device)
+                return {result.device.uuid: result}
+            except ClientConnectorError as err:
```

Comment:
No need to catch these, data update coordinator already handles all aiohttp errors